### PR TITLE
Cluster import

### PR DESCRIPTION
**Pull request by Ecsy**
Related https://github.com/banzaicloud/banzai-cli/issues/82

### CODE REVIEW

File: internal/cli/command/cluster/import.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+}
+
+// NewInstallCommand returns a cobra command for `install` subcommands.
+func NewImportCommand(banzaiCli cli.Cli) *cobra.Command {
+	options := importOptions{}
+
+	cmd := &cobra.Command{
+		Use:     "import",
+		Aliases: []string{"imp"},
+		Short:   "Manage cluster imports",
```
Comment (sagikazarmark) [inline internal/cli/command/cluster/import.go]:
```suggestion
		Short:   "Import an existing cluster",
```

File: internal/cli/command/cluster/import.go [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+	kubeconfig string
+}
+
+// NewInstallCommand returns a cobra command for `install` subcommands.
+func NewImportCommand(banzaiCli cli.Cli) *cobra.Command {
+	options := importOptions{}
+
+	cmd := &cobra.Command{
+		Use:     "import",
+		Aliases: []string{"imp"},
```
Comment (sagikazarmark) [inline internal/cli/command/cluster/import.go]: I don't feel the need to add aliases to every command.

File: internal/cli/command/cluster/import.go [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+		Short:   "Manage cluster imports",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return importCluster(banzaiCli, options)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVarP(&options.name, "name", "n", "", "Name of the cluster")
+	flags.StringVarP(&options.file, "file", "f", "", "Cluster descriptor file")
```
Comment (sagikazarmark) [inline internal/cli/command/cluster/import.go]: This is a kubeconfig, isn't it?

File: internal/cli/command/cluster/import.go [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+	return cmd
+}
+
+func importCluster(banzaiCli cli.Cli, options importOptions) error {
+	pipeline := banzaiCli.Client()
```
Comment (sagikazarmark) [inline internal/cli/command/cluster/import.go]:
```suggestion
	client := banzaiCli.Client()
```

Then you don't have to alias the client package to `pkgPipeline`.

File: internal/cli/command/cluster/import.go [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+		return emperror.Wrap(err, "failed to import Kubernetes cluster")
+	}
+
+	log.Debugf("Kubernetes config secret created with id: %d\n", cluster.Id)
+
+	return nil
+}
+
+func buildInteractiveImportRequest(banzaiCli cli.Cli, options importOptions, orgId int32) (kubeconfig, clusterName string, err error) {
+	_ = banzaiCli
```
Comment (sagikazarmark) [inline internal/cli/command/cluster/import.go]: What's the point of this? If those arguments are not used and the linter complains, you can just replace the variable name in the function definition with the ignore operator.

File: internal/cli/command/cluster/import.go [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+		}
+		options.kubeconfig = base64.StdEncoding.EncodeToString(raw)
+	}
+
+	// Create kubernetes secret
+	req := pkgPipeline.CreateSecretRequest{}
+	req.Name = options.name
+	req.Type = "kubernetes"
+	req.Values = make(map[string]interface{}, 1)
+	req.Values["K8Sconfig"] = options.kubeconfig
```
Comment (orymate) [inline internal/cli/command/cluster/import.go]: A literal for the final request would be easier to read than the assignments

File: internal/cli/command/cluster/import.go [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+	return nil
+}
+
+func buildInteractiveImportRequest(banzaiCli cli.Cli, options importOptions, orgId int32) (kubeconfig, clusterName string, err error) {
+	_ = banzaiCli
+	_ = options
+	_ = orgId
+	// TODO: implement interactive version
+	var fileName = options.file
+	if fileName != "" {
```
Comment (orymate) [inline internal/cli/command/cluster/import.go]: Why dont use the new util function ReadFileOrStdin()?

File: internal/cli/command/cluster/import.go [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+	name       string
+	file       string
+	kubeconfig string
+}
+
+// NewInstallCommand returns a cobra command for `install` subcommands.
+func NewImportCommand(banzaiCli cli.Cli) *cobra.Command {
+	options := importOptions{}
+
+	cmd := &cobra.Command{
```
Comment (orymate) [inline internal/cli/command/cluster/import.go]: Some complete Examples (like the command with kubectl piped as input) would be useful here

File: internal/cli/command/cluster/import.go [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+
+	log.Debugf("Kubernetes config secret created with id: %d\n", cluster.Id)
+
+	return nil
+}
+
+func buildInteractiveImportRequest(banzaiCli cli.Cli, options importOptions, orgId int32) (kubeconfig, clusterName string, err error) {
+	_ = banzaiCli
+	_ = options
+	_ = orgId
```
[the remark this answers is not in the document]
Comment (sagikazarmark) [reply internal/cli/command/cluster/import.go]: See my comment above.

File: internal/cli/command/cluster/import.go [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+	body["cloud"] = "kubernetes"
+	prop := make(map[string]interface{}, 1)
+	prop["kubernetes"] = make(map[string]interface{}, 0)
+	body["properties"] = prop
+
+	if bytes, err := json.MarshalIndent(body, "", "  "); err != nil {
+		log.Errorf("failed to marshal request: %v", err)
+		log.Debugf("Request: %#v", req)
+	} else {
+		_, _ = fmt.Fprintf(os.Stderr, "The current state of the request:\n\n%s\n", bytes)
```
[the remark this answers is not in the document]
Comment (Ecsy) [reply internal/cli/command/cluster/import.go]: copy-paste from cluster create (same as secret delete)

[the remark this answers is not in the document]
Comment (orymate) [reply internal/cli/command/cluster/import.go]: yes, but it's printed there before a final confirmation (that's what the user confirms)

**Pull request merged**